build.sh

```diff
@@ -206,20 +208,30 @@ if [[ ${#_CXX} > 0 ]]; then
     CC_PREFIX="-DCMAKE_CXX_COMPILER=$_CXX -DCMAKE_C_COMPILER=$_CC"
 fi
 
-build_directory="$CHAKRACORE_DIR/BuildLinux/${BUILD_TYPE:0}"
+root_build_directory="$CHAKRACORE_DIR/Build/clang_build/${BUILD_ARCH}_${BUILD_TYPE:0}"
+if [ ! -d "$root_build_directory" ]; then
+    SAFE_RUN `mkdir -p $root_build_directory`
+fi
+
+bin_directory="$root_build_directory/bin"
+if [ ! -d "$bin_directory" ]; then
+    SAFE_RUN `mkdir -p $bin_directory`
+fi
+
+build_directory="$root_build_directory/obj"
 if [ ! -d "$build_directory" ]; then
     SAFE_RUN `mkdir -p $build_directory`
 fi
 
 pushd $build_directory > /dev/null
 
 echo Generating $BUILD_TYPE makefiles
-cmake $CMAKE_GEN $CC_PREFIX $ICU_PATH $STATIC_LIBRARY -DCMAKE_BUILD_TYPE=$BUILD_TYPE $WITHOUT_FEATURES ../..
+cmake $CMAKE_GEN $CC_PREFIX $ICU_PATH $STATIC_LIBRARY -DCMAKE_BUILD_TYPE=${BUILD_TYPE^} $WITHOUT_FEATURES ../../../..
```

Comment:
can we cut this line with `\` to maintain 80 col or something closer to that please?